File: exercism/ruby/high-scores/high_scores.rb
class HighScores
    attr_reader :scores
    def initialize(scores)
        @scores = scores
    end

    def latest
        @scores.last
    end

    def personal_best
        @scores.max
    end

    def personal_top
        @scores.max(3)
    end

    def report 
        "Your latest score was #{latest}. #{sentence_to_best}"
    end

    def sentence_to_best
        diff = personal_best - latest
        if diff > 0 then
            "That's #{diff} short of your personal best!"
        else
            "That's your personal best!"
        end
    end
end

File: exercism/python/high-scores/high_scores.py
def latest(scores):
    return scores[-1]


def personal_best(scores):
    return max(scores)


def personal_top_three(scores):
    res = []
    for _ in range(0, 3):
        if not scores:
            break
        else:
            maximun = max(scores)
            res.append(maximun)
            scores.remove(maximun)
    return res
